examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -162,10 +162,26 @@ CHIP_ERROR CastingServer::SendUserDirectedCommissioningRequest(chip::Transport::
 
 CHIP_ERROR CastingServer::SendUserDirectedCommissioningRequest(Dnssd::DiscoveredNodeData * selectedCommissioner)
 {
+    chip::TransportMgrBase transport = Server::GetInstance().GetTransportManager();
     mUdcInProgress = true;
     // Send User Directed commissioning request
+    int ipIndexToUse = 0;
+    for (int i = 0; i < selectedCommissioner->resolutionData.numIPs; i++)
+    {
+        if (selectedCommissioner->resolutionData.ipAddress[i].IsIPv4())
+        {
+            ipIndexToUse = i;
+            ChipLogProgress(AppServer, "Found iPv4 address at index: %d", ipIndexToUse);
+            break;
+        }
+
+        if (i == (selectedCommissioner->resolutionData.numIPs - 1))
+        {
+            ChipLogProgress(AppServer, "Could not find an iPv4 address, defaulting to the first address in IP list");
+        }
+    }
```

Comment:
Can you move this to a private function like selectIPAddressForUDCRequest() or something like that, add some documentation at the top of it to say why we are doing it and just call it from here? Because why we are selecting an IPv4 address is not immediately obvious from looking at the code.

Reply:
Done